src/main/java/fi/aalto/cs/apluscourses/dal/APlusExerciseDataSource.java

```diff
@@ -137,21 +141,31 @@ public Points getPoints(@NotNull Course course, @NotNull Authentication authenti
     return parser.parsePoints(response);
   }
 
+  @Override
+  @NotNull
+  public SubmissionResult getSubmissionResult(@NotNull String submissionUrl,
```

Comment:
nit: other methods in this class construct the URL to be called inside them, and this one does not

Reply:
That's a good point and I thought about it as well.

However, the submission API works such that the response header has the URL to the submission that was created. Therefore we don't need to construct the URL here at all since it's given in the submission API response.